buildenv/jenkins/jobs/infrastructure/artifactory_cleanup.groovy

```diff
@@ -0,0 +1,131 @@
+
+        if (params.JOB_TYPE == 'Time'){
+            cleanupTime(ARTIFACTORY_SERVER, ARTIFACTORY_REPO, ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS)
+        } else if (params.JOB_TYPE == 'Number of Builds'){
+            if (params.JOB_TO_CHECK){
+                cleanupBuilds(ARTIFACTORY_SERVER, ARTIFACTORY_REPO, JENKINS_DOMAIN, params.JOB_TO_CHECK, ARTIFACTORY_NUM_ARTIFACTS)
+            } else {
+                error 'Please input a job to cleanup'
+            }
+        }
+    }
+}
+
+def cleanupBuilds(artifactory_server, artifactory_repo, jenkins_domain, jobToCheck, artifactory_num_artifacts){
+    // This parameter is originally a string and needs to be casted as an Integer
+    def artifactory_num_artifacts = artifactory_num_artifacts as Integer 
+
+    stage('Discover Stored Artifacts'){
+        echo "Cleaning up ${jobToCheck}"
+        echo "Keeping the latest ${artifactory_num_artifacts} builds"
+
+        def request = httpRequest authentication: env.ARTIFACTORY_CREDS, consoleLogResponseBody: true, url: "${artifactory_server}/api/storage/${artifactory_repo}/${jenkins_domain}/${jobToCheck}"
+
+        data = readJSON text: request.getContent()
+        numberOfArtifacts = data.children.size()
+        echo "There are ${numberOfArtifacts} builds"
+    } 
+    stage('Delete Old Artifacts'){
+        if (numberOfArtifacts > artifactory_num_artifacts){
+            def folderNames = getFolderNumbers(data.children.uri)
+
+            for(i=0; i  < (numberOfArtifacts - artifactory_num_artifacts); i++){
+                echo "Deleting Build #${folderNames[i]}"
+                httpRequest authentication: env.ARTIFACTORY_CREDS, httpMode: 'DELETE', consoleLogResponseBody: true, url: "${artifactory_server}/${env.ARTIFACTORY_REPO}/${jobToCheck}/${folderNames[i]}"
+            }
+        } else {
+            echo 'There are no artifacts to delete'
+        } 
+    }
+}
+
+def getFolderNumbers(someVariable){
+    def folderNumbers = []
+    someVariable.each{
+        folderNumbers.add(it.minus('/') as int )
+    }
+    return folderNumbers.sort()
+}
+
+def cleanupTime(artifactory_server, artifactory_repo , artifactory_days_to_keep_artifacts){
+    stage('Discover Old Artifacts') {
+        // This parameter is originally a string and needs to be casted as an Integer
+        artifactory_days_to_keep_artifacts = artifactory_days_to_keep_artifacts as Integer
+        def date = new Date()
+        def current_time = date.getTime()
+        def created_before_time = ate.minus(artifactory_days_to_keep_artifacts).getTime()
+        echo "Getting all artifacts over ${artifactory_days_to_keep_artifacts} days old"
+        def request = httpRequest authentication: env.ARTIFACTORY_CREDS, consoleLogResponseBody: true, validResponseCodes: '200,404', url: "${artifactory_server}/artifactory/api/search/usage?notUsedSince=${current_time}&createdBefore=${created_before_time}&repos=${artifactory_repo}"
```

Comment:
I'm guessing we can't remove `notUsedSince`?

Reply:
I was unable to remove that